### PR TITLE
Support setting the new relic transaction name

### PR DESCRIPTION
**Pull request by rmosolgo**
This adds an option to new relic tracer to set the transaction name for all queries.

### CODE REVIEW

File: lib/graphql/tracing/new_relic_tracing.rb [hunk 1]
```diff
@@ -14,7 +14,29 @@ class NewRelicTracing < PlatformTracing
         "execute_query_lazy" => "GraphQL/execute",
       }
 
+      # @param set_transaction_name [Boolean] If true, the GraphQL operation name will be used as the transaction name.
+      #   This is not advised if you run more than one query per HTTP request, for example, with `graphql-client` or multiplexing.
+      def initialize(set_transaction_name: false)
+
+        @set_transaction_name = true
```
Comment (akshah123) [inline lib/graphql/tracing/new_relic_tracing.rb]: looks like this might always set it to `true`.

Comment (rmosolgo) [reply lib/graphql/tracing/new_relic_tracing.rb]: 🤦‍♂️ 🤦‍♂️ 🤦‍♂️ 🤦‍♂️ 🤦‍♂️ 🤦‍♂️ 🤦‍♂️ 🤦‍♂️ 🤦‍♂️ 🤦‍♂️ 🤦‍♂️ 🤦‍♂️ 🤦‍♂️ 🤦‍♂️ 🤦‍♂️ 🤦‍♂️ 🤦‍♂️ 🤦‍♂️ 🤦‍♂️ 🤦‍♂️ 🤦‍♂️ 🤦‍♂️ 🤦‍♂️ 🤦‍♂️ 🤦‍♂️ 🤦‍♂️ 🤦‍♂️

File: lib/graphql/tracing/new_relic_tracing.rb [hunk 2]
```diff
@@ -14,7 +14,29 @@ class NewRelicTracing < PlatformTracing
...
+      end
+
       def platform_trace(platform_key, key, data)
+        if @set_transaction_name && platform_key == "execute_query"
```
[the remark this answers is not in the document]
Comment (rmosolgo) [reply lib/graphql/tracing/new_relic_tracing.rb]: yeah :S I've been putting off proper unit tests for these platform tracers, but two fatal errors in a row means I can't afford to skip them anymore 😆  thanks!

**Comment by rmosolgo** [comment]
Thanks again for the reviews, I've pushed some proper tests and fixed a few issues, and added a simple per-query override. @obrie , thanks for sharing your advanced use case -- I'd like to publish these small improvements as-is, and consider that more sophisticated implementation later, if more folks express interest. It sounds like nice work!

**Pull request merged**